public/app/features/inspector/InspectDataTab.tsx

```diff
@@ -167,18 +167,17 @@ export class InspectDataTab extends PureComponent<Props, State> {
 
   renderDataOptions(dataFrames: DataFrame[]) {
     const { options, onOptionsChange, panel, data } = this.props;
-    if (!panel || !onOptionsChange) {
-      return null;
-    }
     const { transformId, transformationOptions, selectedDataFrame } = this.state;
 
     const styles = getPanelInspectorStyles();
 
-    const panelTransformations = panel.getTransformations();
+    const panelTransformations = panel?.getTransformations();
     const showPanelTransformationsOption =
-      panelTransformations && panelTransformations.length > 0 && (transformId as any) !== 'join by time';
-    const showFieldConfigsOption = !panel.plugin?.fieldConfigRegistry.isEmpty();
-    const showDataOptions = showPanelTransformationsOption || showFieldConfigsOption;
+      onOptionsChange &&
+      panelTransformations &&
+      panelTransformations.length > 0 &&
+      (transformId as any) !== 'join by time';
```

Comment:
nit: I think this can be simplified into:
```suggestion
      onOptionsChange &&
      Boolean(panelTransformations?.length) &&
      (transformId as any) !== 'join by time';
```